credentials/java/com/google/auth/ApiKeyCredentials.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2024, Google Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are
+ * met:
+ *
+ *    * Redistributions of source code must retain the above copyright
+ * notice, this list of conditions and the following disclaimer.
+ *    * Redistributions in binary form must reproduce the above
+ * copyright notice, this list of conditions and the following disclaimer
+ * in the documentation and/or other materials provided with the
+ * distribution.
+ *
+ *    * Neither the name of Google Inc. nor the names of its
+ * contributors may be used to endorse or promote products derived from
+ * this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.google.auth;
+
+import com.google.api.client.util.*;
+import java.io.IOException;
+import java.net.URI;
+import java.util.*;
+
+/**
+ * Credentials class for calling Google APIs using an API key.
+ *
```

Comment:
Could you add a small snippet in the javadocs (or perhaps anywhere else) explaining why this extends directly from Credentials and not GoogleCredentials/ Oauth2Credentials? 

I believe the original reason is that Api Keys do not have a concept of Access Tokens, but not sure if there were additional reasons behind this.

Reply:
I added it in the PR description as its seems a design decision and wouldn't be part of java docs, but happy to add there if think it would be helpful

Reply:
PR Description does help, thanks! I think it may be a bit more complicated/ difficult to trace back the decision to this PR (especially if more PRs/ changes come in). I think that either a short blurb in the comments or perhaps in the original design doc would be easier to find.

Reply:
ok added to java docs